### PR TITLE
#32: Template renaming should not be exclusive to components

### PR DESCRIPTION
**Pull request by alxndr13**
first draft of a possible implementation of the renameConfig stuff

### CODE REVIEW

File: template.go [hunk 1]
```diff
@@ -73,11 +73,22 @@ func NewTemplater(fileSystem afero.Fs, templateRootPath, outputRootPath string,
...
+				fmt.Println("new output:", rename.Filename)
+				return t.execute(template, data, rename.Filename, allowNoValue, nil)
+			}
+		}
+	}
+	return t.execute(template, data, template.Path, allowNoValue, nil)
 }
 
-func (t *Templater) execute(template *TemplateFile, data any, targetPath string, allowNoValue bool) error {
+func (t *Templater) execute(template *TemplateFile, data any, targetPath string, allowNoValue bool, renameConfig []RenameConfig) error {
```
Comment (lukasjarosch) [inline template.go:91]: Is adding `renameConfig` still required? The targetPath should be sufficient and it seems that you're not using it.

File: inventory.go [hunk 2]
```diff
@@ -165,6 +165,33 @@ func (inv *Inventory) GetCopyConfigs(targetName string) ([]CopyConfig, error) {
...
+
+	for _, class := range usedClasses {
+		if class.Configuration.IsSet() {
+			renameConfigs = append(renameConfigs, class.Configuration.Renames...)
+		}
+	}
+
+	return renameConfigs, nil
+}
+
```
[the remark this answers is not in the document]
Comment (alxndr13) [reply inventory.go:194]: I will rebase upon that branch, so we got that included right here. :)

File: template.go (hunk 1 above)
Comment (alxndr13) [reply template.go:91]: nope, you're right. That was the first draft, will remove - thanks for the headsup!

**Comment by lukasjarosch** [comment]
King! ❤️

**Pull request merged**